src/api.js

```diff
@@ -135,6 +136,45 @@ export default class Api {
   }
 
   /**
+   * Process a batch request response.
+   *
+   * @param  {Object}  results          The results object.
+   * @param  {Array}   records          The initial records list.
+   * @param  {Number}  response.status  The response HTTP status.
+   * @param  {Object}  response.json    The response JSON body.
+   * @param  {Headers} response.headers The response headers object.
+   * @return {Promise}
+   */
+  _processBatchResponses(results, records, {status, json, headers}) {
+    // Handle individual batch subrequests responses
+    json.responses.forEach((response, index) => {
+      // TODO: handle 409 when unicity rule is violated (ex. POST with
+      // existing id, unique field, etc.)
+      if (response.status && response.status >= 200 && response.status < 400) {
+        results.published.push(response.body.data);
+      } else if (response.status === 404) {
+        results.skipped.push(response.body);
+      } else if (response.status === 412) {
+        results.conflicts.push({
+          type: "outgoing",
+          local: records[index],
```

Comment:
PS: for this very method, the indexes of passed `records` can't change really; though the chunks being processed concurrently is a very valid issue.